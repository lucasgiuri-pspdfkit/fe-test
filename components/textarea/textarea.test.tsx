import Textarea from "./textarea";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Textarea", () => {
  it("should render the textarea component", () => {
    const onChange = jest.fn();
    render(
      <Textarea
        placeholder="Enter text"
        value="some value"
        onChange={onChange}
      />
    );
    const textarea = screen.getByPlaceholderText("Enter text");
    expect(textarea).toBeInTheDocument();
  });

  it("Snapshot with isRequired", () => {
    const onChange = jest.fn();
    const { container } = render(
      <Textarea placeholder="" value="" onChange={onChange} isRequired />
    );
    expect(container).toMatchSnapshot();
  });
});
