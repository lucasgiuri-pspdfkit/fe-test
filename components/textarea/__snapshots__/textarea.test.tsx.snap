// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Snapshot with isRequired 1`] = `
<div>
  <textarea
    class="w-full h-[161px] p-3 text-gray-500 border border-gray-200 placeholder-gray-200 focus:ring-0 outline-gray-500 rounded-md"
    placeholder=""
    required=""
  />
</div>
`;
